src/analysis/souffle/tags.dl

```diff
@@ -0,0 +1,41 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+// Definition of the base tag types that are used by this dataflow analysis.
+
+#ifndef SRC_ANALYSIS_SOUFFLE_TAGS_DL_
+#define SRC_ANALYSIS_SOUFFLE_TAGS_DL_
+
+// A symbol representing a taint tag. SecrecyTags indicate that the data upon which they are
+// annotated potentially contain information that should be handled with some sort of special care
+// and should not be exposed to some outside group. The conservative approximation for SecrecyTags
+// is, when you are in doubt whether a particular datum should have this tag, annotate it. This
+// causes these tags to spread like taint.
+.type SecrecyTag <: symbol
```

Comment:
The term confidentiality is used more often than secrecy in IFC literature (although the Flume paper happens to be one that uses secrecy :) )